Neos.ContentRepository/Classes/Domain/Projection/Content/NodeInterface.php

```diff
@@ -0,0 +1,134 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Domain\Projection\Content;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Domain\Model\NodeType;
+use Neos\ContentRepository\Domain\ValueObject\NodeIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\ContentStreamIdentifier;
+use Neos\ContentRepository\DimensionSpace\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Domain\ValueObject\NodeAggregateIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\NodeName;
+use Neos\ContentRepository\Domain\ValueObject\NodeTypeName;
+use Neos\Cache\CacheAwareInterface;
+use Neos\ContentRepository\Domain\ValueObject\PropertyCollectionInterface;
+
+/**
+ * This is a NEW interface, introduced in Neos 4.3.
+ *
+ * The new Event-Sourced core NodeInterface used for READING. It contains only information
+ * local to a node; i.e. all properties in this interface can be accessed extremely fast.
+ *
+ * Most likely, you want to use {@see TraversableNodeInterface} instead, as it is more
+ * convenient to use for end-users, as it also contains node traversal operations.
+ *
+ * The NodeInterface is *immutable*, meaning its contents never change after creation.
+ * It is *only used for reading*.
+ *
+ * Starting with version 5.0 (when backed by the Event Sourced CR), it is
+ * *completely detached from storage*; so it will not auto-update after a property changed in
+ * storage.
+ */
+interface NodeInterface extends CacheAwareInterface
+{
+    /**
+     * Whether or not this node is the root of the graph, i.e. has no parent node
+     *
+     * @return bool
+     */
+    public function isRoot(): bool;
+
+    /**
+     * @return ContentStreamIdentifier
+     */
+    public function getContentStreamIdentifier(): ContentStreamIdentifier;
+
+    /**
+     * returns the DimensionSpacePoint the node was *requested in*, i.e. one of the DimensionSpacePoints
+     * this node is visible in. If you need the DimensionSpacePoint where the node is actually at home,
+     * see getOriginDimensionSpacePoint()
+     *
+     * @return DimensionSpacePoint
+     */
+    public function getDimensionSpacePoint(): DimensionSpacePoint;
```

Comment:
Can we skip this until we established that we really need it? This is part of the requested subgraph's identity and not necessary to identify a node. In fact this may lead to multiple instances of the exact same node in different DSPs. I'm fine with multipe TraversableNode-Wrappers for that case as they wrap different subgraphs but I don't see any use for this here

Reply:
Nope, needed in various places in the core currently. Multuple instances of the exact same node will happen anyways; as there is no identity map involved.

I am fine with discussing this later and if needed drop this property again; but so far it is needed in the core - and we'd need a working PR which works without it IMHO.

All the best, Sebastian

Reply:
Of course, if the subgraph stays in, we could remove it.

Reply:
see, below, will also look into it